app/api-v1/controllers/Attachment/index.js

```diff
@@ -1,5 +1,72 @@
+const { parseAccept } = require('../../../utils/parseAcceptHeader')
+const logger = require('../../../utils/Logger')
+
+const db = require('../../../db')
+
+const { createAttachmentFromFile, createAttachment } = require('../Attachment/helpers')
+const { InternalError, NotFoundError, BadRequestError } = require('../../../utils/errors')
+
 module.exports = {
-  getAll: async function () {
-    return { status: 500, response: { message: 'Not Implemented' } }
+  get: async (req) => {
+    const { id } = req.params
+    if (!id) throw new BadRequestError('missing params')
+
+    const [attachment] = await db.getAttachment(req.params.id)
+    if (!attachment) throw new NotFoundError('Attachment Not Found')
+    const orderedAccept = parseAccept(req.headers.accept)
+
+    for (const mimeType of orderedAccept) {
+      if (mimeType === 'application/octet-stream' && attachment.filename.includes('json'))
+        return {
+          status: 406,
+          response: 'some error msg',
+        }
+      if (mimeType === 'application/json' || mimeType === 'application/*' || mimeType === '*/*') {
+        const json = JSON.parse(attachment.binary_blob)
+        //It looks like the validator is still the issue, I get a status f 200 befre validatioo
+        return {
+          status: 200,
+          response: json,
+        }
+      }
+      if (mimeType === 'application/octet-stream') {
+        return {
+          status: 201,
+          headers: {
+            immutable: true,
+            maxAge: 365 * 24 * 60 * 60 * 1000,
+            'content-disposition': `attachment; filename="${attachment.filename}"`,
+            'access-control-expose-headers': 'content-disposition',
+            'content-type': 'application/octet-stream',
+          },
+          response: attachment.binary_blob.toString(),
```

Comment:
This is almost certainly wrong right? It'll represent the response as a utf-8 string which it's not, it's binary. Is there a reason we can't just pass the buffer?

Reply:
At the moment the validator is returning a string. I'll have a look and see if binary is an accepted schema type.

Reply:
This is irrelevant. The response needs to be binary. That's spec.

Reply:
If the validator can't support that then we shouldn't be using the validator as it doesn't support necessary functionality